### PR TITLE
Add TestGesture

### LINKED ISSUE #1855: TestPointer changes
Driving a pointer in a test right now requires pumping, continually giving the original location, etc.

Instead the API should look something like:

```dart
TestGesture gesture = tester.startGesture(downLocation);
gesture.moveTo(newPoint1);
gesture.moveTo(newPoint2);
gesture.moveTo(newPoint3);
gesture.up(); // or gesture.cancel();
```

moveTo would dispatch a pointer move event with the new point, at downLocation, then pump.

gesture would also expose the TestPointer.

moveTo() would take optional argument "pump: false" or maybe "duration" which is how long to pump (and null would mean don't pump).

### PR DESCRIPTION
**Pull request by abarth**
This helper makes it easier to write correct tests that involve
gestures.

Fixes #1855

### CODE REVIEW

**Comment by abarth** [comment]
@Hixie

File: packages/flutter_test/lib/src/instrumentation.dart [hunk 1]
```diff
@@ -161,6 +161,14 @@ class Instrumentation {
...
   }
 
+  TestGesture startGesture(Point downLocation, { int pointer: 1 }) {
+    TestPointer p = new TestPointer(pointer);
+    HitTestResult result = _hitTest(downLocation);
+    _dispatchEvent(p.down(downLocation), result);
+    return new TestGesture._(this, result, p);
+  }
+
+  @deprecated // Use startGesture instead.
```
Comment (Hixie) [inline packages/flutter_test/lib/src/instrumentation.dart]: `@deprecated('soon. Use startGesture instead.')`

**Comment by Hixie** [comment]
lgtm

**Comment by Hixie** [comment]
We should also have `TestGesture.cancel()`.

**Comment by abarth** [comment]
Done.

**Pull request merged**